### PR TITLE
Windows fixes

### PR DESCRIPTION
**Pull request by margaretkoehler**
Resolve the compilation problem for windows following sofa_cmake changes (similar to PR #41)
Add old fixes for windows compilation with CommunicationController

### CODE REVIEW

**Comment by EulalieCoevoet** [comment]
Thanks Margaret! Looks fine to me. Works on windows. 
I don't understand the CI error with ubuntu though... For me it doesn't seem related to this PR...

**Comment by EulalieCoevoet** [comment]
@marques-bruno @damienmarchal should we merge regardless of the CI? or can someone check the problem with ubuntu...

File: src/SoftRobots/component/controller/CommunicationController.inl [hunk 1]
```diff
@@ -318,22 +318,22 @@ void CommunicationController<DataTypes>::sendData()
...
 
     zmq::message_t message;
-    unsigned int messageSize;
     bool status = m_socket->recv(&message);
+
     if(status)
     {
-        char messageChar[messageSize];
-        memcpy(&messageChar, message.data(), messageSize);
+        char* messageChar = new char[message.size()];
```
Comment (damienmarchal) [inline src/SoftRobots/component/controller/CommunicationController.inl]: Isn't there a memory leak here ? I don't see the corresponding delete.

**Comment by damienmarchal** [comment]
[ci-build]

File: src/SoftRobots/component/controller/CommunicationController.inl (hunk 1 above)
Comment (margaretkoehler) [reply src/SoftRobots/component/controller/CommunicationController.inl]: Good catch, @damienmarchal ! I think I fixed it.

**Comment by marques-bruno** [comment]
The ci builds so it's ok... Sadly we can't (yet) rely on the ci to properly run the tests, hence the crashes. I'll fix it ASAP, hopefully... Maybe next Friday

**Comment by EulalieCoevoet** [comment]
This has been merged in another MR. Thanks Margaret.

**Pull request closed without merging**